application/src/main/java/org/togetherjava/tjbot/commands/meme/MemeListener.java

```diff
@@ -0,0 +1,65 @@
+package org.togetherjava.tjbot.commands.meme;
+
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.requests.RestAction;
+import net.dv8tion.jda.api.requests.restaction.AuditableRestAction;
+import org.jetbrains.annotations.NotNull;
+import org.togetherjava.tjbot.commands.MessageReceiverAdapter;
+import org.togetherjava.tjbot.config.Config;
+
+import java.util.regex.Pattern;
+
+/**
+ * Listener that receives all sent messages from the Meme channel, checks if the message has media
+ * attched.
+ * <p>
+ * If there was no media attached, delete the messagen and send the User a DM, telling what they did
+ * wrong.
+ */
+public final class MemeListener extends MessageReceiverAdapter {
```

Comment:
similar to the config entry, id actually like this whole class to change into a generalized "media only channel listener". might also want to add `#educational_media` to it and maybe more in the future. sorry for the inconvenience, but it think its a low hanging fruit we should harvest.

Reply:
renamed to MediaOnlyChannelListener